File: ruby/raindrops/raindrops.rb
class Raindrops
  VERSION = 1

  def self.convert(number)
    response = check_primes(number)
    if response == []
      response << number
    end
    response.join("")
  end

  def self.check_primes(number)
    raindrops_hash = {3 => "Pling", 5 => "Plang", 7 => "Plong"}
    response = []
    raindrops_hash.keys.each do |key|
      if number % key == 0
        response << raindrops_hash[key]
      end
    end
    response
  end
end

File: php/raindrops/raindrops.php
<?php

function raindrops($num)
{
  $result = "";

  $raindrops_array = [
    3 => "Pling",
    5 => "Plang",
    7 => "Plong"
  ];

  foreach ($raindrops_array as $key => $value) {
    if ($num%$key == 0)
      $result .= $value;
  }

  if (strlen($result) == 0)
    return "$num";
  else
    return $result;
}
